docs: migrate 'metrics and statistics' (#5372)
canonical/snapcraft: docs/reference/metrics.rst

.. _reference-metrics:

Metrics
=======

This page lists all supported statistics that stores gather about snaps.

> All metrics are generated daily. Those with seven-day windows provide the average
> averages.

Reviewer: I think this is a typo:
```suggestion
All metrics are generated daily. Those with seven-day windows provide the averages.
```

Author: Whoops: [98d19bc](https://github.com/canonical/snapcraft/pull/5372/commits/98d19bc604c09abf310ee8e6ea100b917829f04d)

> All metrics are generated daily. Those with seven-day windows provide daily figures, but
> these are not raw values, but the daily average.

Reviewer 2: ```suggestion
All metrics are generated daily. Those with seven-day windows provide daily figures, but
these are not raw values; rather, they are the daily averages.
```
Original sentence didn't sound quite right to me. I'm sure you could reword it any number of ways, so take this suggestion lightly.

Author: Yeah, this one took me a few tries. Awkward to describe: 1bf114a

.. list-table::
    :header-rows: 1
    :widths: 1 2

    * - Metric
      - Data
    * - ``daily_device_change``

>       - Contains the 3 series representing the number of new, continued and lost devices with the given snap installed compared to the previous day.

Reviewer 2: ```suggestion
      - Contains the three series representing the number of new, continued and lost devices with the given snap installed compared to the previous day.
```
This is the only description that uses a numeral.

Author: 1bf114a

    * - ``installed_base_by_channel``

>       - Contains one series per channel representing the number of devices with the
>         given snap installed, channels with no data across the entire interval are
>         omitted.

Reviewer 2: ```suggestion
      - Contains one series per channel representing the number of devices with the
        given snap installed. Channels with no data across the entire interval are
        omitted.
```

Author: 1bf114a

    * - ``installed_base_by_country``
      - Contains one series per country representing the number of devices with the
        given snap installed.
    * - ``installed_base_by_operating_system``

>       - Contains one series per operating_system representing the number of devices with

Reviewer: ```suggestion
      - Contains one series per operating system representing the number of devices with
```

Author: It wasn't me :eyes: [98d19bc](https://github.com/canonical/snapcraft/pull/5372/commits/98d19bc604c09abf310ee8e6ea100b917829f04d)

        the given snap installed.
    * - ``installed_base_by_version``

>       - Contains one series per version representing the number of devices with the
>         given snap installed.

Reviewer 2: ```suggestion
      - Contains one series per version representing the number of devices with that
        snap version installed.
```
I may just be misunderstanding this one.

Author: 1bf114a

    * - ``installed_base_by_architecture``
      - Contains one series per architecture representing the number of devices with the
        given snap installed.
    * - ``weekly_device_change``

>       - Similar to the ``daily_device_change`` metric but operates on a seven-day
>         window. In other words, contains the number of devices that were seen during the
>         last seven days but not in the previous seven day and so on for continued and lost.

Reviewer 2: ```suggestion
      - Similar to the ``daily_device_change`` metric but operates on a seven-day
        window.
```
I don't think this requires the extra explanation.

    * - ``weekly_installed_base_by_channel``
      - Similar to the ``installed_base_by_channel`` metric but operates in a seven-day
        window.
    * - ``weekly_installed_base_by_country``
      - Similar to the ``installed_base_by_country`` metric but operates in a seven-day
        window.
    * - ``weekly_installed_base_by_operating_system``
      - Similar to the ``installed_base_by_operating_system`` metric but operates in a
        seven-day window.
    * - ``weekly_installed_base_by_version``
      - Similar to the ``installed_base_by_version`` metric but operates in a seven-day
        window.
    * - ``weekly_installed_base_by_architecture``
      - Similar to the ``installed_base_by_architecture`` metric but operates in a

>         seven- ay window.

Reviewer: ```suggestion
        seven-day window.
```

Author: Oopsie: [98d19bc](https://github.com/canonical/snapcraft/pull/5372/commits/98d19bc604c09abf310ee8e6ea100b917829f04d)
